src/main/java/com/iexec/core/replicate/ReplicatesService.java

```diff
@@ -366,27 +451,25 @@ private boolean canUpdateToBlockchainSuccess(String chainTaskId, Replicate repli
         return true;
     }
 
-    private boolean canUpdateToUploadSuccess(String chainTaskId, Replicate replicate,
-                                             ReplicateStatusUpdate statusUpdate) {
+    private boolean canUpdateToUploadSuccess(String chainTaskId,
+                                             Replicate replicate,
+                                             ReplicateStatusUpdate statusUpdate,
+                                             UpdateReplicateStatusArgs updateReplicateStatusArgs) {
         ReplicateStatus newStatus = statusUpdate.getStatus();
-        ReplicateStatusDetails details = statusUpdate.getDetails();
 
         if (!verifyStatus(chainTaskId, replicate.getWalletAddress(), newStatus)) {
             log.error("Cannot update replicate, status is not correct",
                     getStatusUpdateLogs(chainTaskId, replicate, statusUpdate));
             return false;
         }
 
-        if (details == null || details.getResultLink() == null || details.getResultLink().isEmpty()) {
+        if (updateReplicateStatusArgs.getResultLink() == null || updateReplicateStatusArgs.getResultLink().isEmpty()) {
```

Comment:
StringUtils.isEmpty

Reply:
Thanks! This is fixed in the next commit.

Reply:
Are you talking about `chainTaskId`, `replicate`, `statusUpdate` and `updateReplicateStatusArgs`? I don't think they should be checked there. It would be better to check them as soon as they are initialized in `replicatesService`.
However, it was not done before my PR. Do you really think it is necessary?